data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java

```diff
@@ -99,6 +101,19 @@ private void buildPipelineFromConfiguration(
             final List<List<Processor>> processorSets = pipelineConfiguration.getProcessorPluginSettings().stream()
                     .map(this::newProcessor)
                     .collect(Collectors.toList());
+
+            final List<List<Processor>> decoratedProcessorSets = processorSets.stream()
+                    .map(processorSet -> processorSet.stream()
+                            .map(processor -> {
+                                if (processor instanceof RequiresPeerForwarding) {
+//                                    TODO: Create buffer per stateful processor and store map of processor, buffer
+                                    return new PeerForwardingProcessorDecorator(processor);
```

Comment:
Now that we are wrapping actual processors, this PR must also update the `PeerForwardingProcessDecorator` to correctly call the inner processors. Thus, it should be a decorator that adds no value yet.

e.g.

```
@Override
public boolean isReadyForShutdown() {
    return innerProcessor.isReadyForShutdown();
}
```

Please implement all methods to call the inner processor.

And add unit tests for these.